aws/resource_aws_spot_fleet_request.go

```diff
@@ -319,6 +319,12 @@ func resourceAwsSpotFleetRequest() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validateRFC3339TimeString,
 			},
+			"fleet_type": {
```

Comment:
Is there a particular reason to call this attribute `fleet_type` and not `type` to match the API?

We can also validate the input via:

```go
ValidateFunc: validation.StringInSlice([]string{
  ec2.FleetTypeMaintain,
  ec2.FleetTypeRequest,
}, false),
```

Reply:
We generally do this because its reduces any differences between the [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/ec2/request-spot-fleet.html), [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-spotfleet-spotfleetrequestconfigdata.html), and other SDK implementations. Its probably fine though.